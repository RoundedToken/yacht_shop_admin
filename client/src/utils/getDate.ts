export function getDate(isoDate: string): string {
    const date = new Date(isoDate);

    return date.toLocaleString('default', {
        day: '2-digit',
        month: 'short',
        year: 'numeric',
        hour: 'numeric',
        minute: 'numeric',
        hour12: false,
    });
}
