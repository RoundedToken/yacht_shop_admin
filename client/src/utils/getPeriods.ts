const toIso = (mils: number) => {
    const now = Date.now();

    return new Date(now - mils).toISOString();
};
const getThisWeek = () => {
    const today = new Date();

    const dayOfWeek = today.getDay() || 7;

    const monday = new Date(today.getTime() - (dayOfWeek - 1) * 24 * 60 * 60 * 1000);

    return monday.toISOString();
};

const getThisMonth = () => {
    const today = new Date();

    const year = today.getFullYear();
    const month = today.getMonth();

    const firstDay = new Date(year, month, 1, 23, 59);

    return firstDay.toISOString();
};

const getThisYear = () => {
    const today = new Date();

    const year = today.getFullYear();

    const firstDay = new Date(year, 0, 1, 23, 59);

    return firstDay.toISOString();
};

export const getPeriods = () => {
    const today = new Date().toISOString();
    const thisWeek = getThisWeek();
    const week = toIso(1000 * 60 * 60 * 24 * 7);
    const month = toIso(1000 * 60 * 60 * 24 * 30);
    const thisMonth = getThisMonth();
    const thisYear = getThisYear();
    const year = toIso(1000 * 60 * 60 * 24 * 365);

    return { week, month, year, today, thisWeek, thisMonth, thisYear };
};
