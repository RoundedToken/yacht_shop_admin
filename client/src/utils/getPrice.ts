export const getPrice = (price: number) => {
    const formatter = new Intl.NumberFormat('et-EE', {
        style: 'currency',
        currency: 'EUR',
    });
    return formatter.format(price);
};
