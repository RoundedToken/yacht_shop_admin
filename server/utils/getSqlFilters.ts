import { IFilters } from '../types';

export const getSqlFilters = (filters: string, isProducts: boolean = false) => {
    const filtersObj = JSON.parse(filters) as IFilters;
    const productId = filtersObj.product;

    const filterStr =
        Object.entries(filtersObj)
            .filter((filter) => filter[1] !== null && filter[1] !== '')
            .map((filter) => {
                const key = filter[0];
                const value = filter[1];

                if (key === 'dateFrom') {
                    return `ord.DateEnter >= CAST('${value}T00:00:00' AS datetime)`;
                } else if (key === 'dateTo') {
                    return `ord.DateEnter <= CAST('${value}T23:59:59' AS datetime)`;
                } else if (key === 'status') {
                    return `ord.status IN (${value.join(',')})`;
                } else if (key === 'isPaid') {
                    return `ord.paid = ${value ? 1 : 0}`;
                } else if (key === 'product') {
                    if (isProducts) return '1=1';
                    else return `ordStr.tovar = ${value}`;
                } else if (key === 'client') {
                    return `ord.clientid = ${value}`;
                } else {
                    return `STUFF((SELECT DISTINCT ', ' + oplata.type  
                    FROM Oplata 
                    WHERE Oplata.orderid = ord.orderid  
                    FOR XML PATH('')), 1, 2, '') LIKE '%${value}%'`;
                }
            })
            .join(' AND ') || '1=1';

    return { filterStr, productId };
};
