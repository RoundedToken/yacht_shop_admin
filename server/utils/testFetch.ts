export class testFetch {
    private endpoint: string;

    constructor(endpoint: string) {
        this.endpoint = endpoint;
    }

    async getData(params: Record<string, string>) {
        const baseUrl = 'http://192.168.1.37:5000/api';
        const url = baseUrl + this.endpoint + '?' + new URLSearchParams(params);

        const response = await fetch(url);

        return await response.json();
    }
}
