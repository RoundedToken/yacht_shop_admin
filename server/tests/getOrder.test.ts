import { describe, expect, test } from 'bun:test';
import { testFetch } from '../utils/testFetch';

describe('Get Order', () => {
    const getOrder = new testFetch('/get_order');

    test('Order 204811', async () => {
        const params = { id: '204811' };
        const data = await getOrder.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Order 204453', async () => {
        const params = { id: '204453' };
        const data = await getOrder.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Order 204426', async () => {
        const params = { id: '204426' };
        const data = await getOrder.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Order 205162', async () => {
        const params = { id: '205162' };
        const data = await getOrder.getData(params);

        expect(data).toMatchSnapshot();
    });
});
