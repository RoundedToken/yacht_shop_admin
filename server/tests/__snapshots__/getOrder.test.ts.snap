// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Order Order 204811 1`] = `
{
  "order": {
    "allSum": -824.1996,
    "clientName": "Nautix",
    "date": "2023-04-29T17:50:00.000Z",
    "id": 204811,
    "isPaid": false,
    "paidSum": 824.1996,
    "positionCount": 8,
    "productsCount": 48,
    "productsSum": -824.1996,
    "status": "покупка",
  },
  "payments": [
    {
      "date": "2023-04-30T15:42:00.000Z",
      "id": 84717,
      "sum": 824.1996,
      "type": "Б/н",
    },
  ],
  "products": [
    {
      "brand": "Nautix",
      "category": "Грунты",
      "categoryId": 161,
      "count": 6,
      "id": 8186,
      "name": "Двухкомпонентный грунт серый PE 0.75 л",
      "price": 17.9828,
      "sum": 107.8968,
    },
    {
      "brand": "Nautix",
      "category": "Краски для палубы",
      "categoryId": 168,
      "count": 6,
      "id": 8260,
      "name": "Нескользящая двухкомпонентная краска Nautix Grip 2 0.5 кг бесцветная",
      "price": 14.8072,
      "sum": 88.8432,
    },
    {
      "brand": "Nautix",
      "category": "Полироли, полировальные круги",
      "categoryId": 203,
      "count": 6,
      "id": 8282,
      "name": "Полироль Polish Boat Rubbing 1.0 кг",
      "price": 15.8296,
      "sum": 94.9776,
    },
    {
      "brand": "Nautix",
      "category": "Необрастающие краски сезонные",
      "categoryId": 587,
      "count": 6,
      "id": 8757,
      "name": "Необрастающая краска Marin Navy Blue 0.75 l",
      "price": 16.0702,
      "sum": 96.4212,
    },
    {
      "brand": "Nautix",
      "category": "Необрастающие краски твердые",
      "categoryId": 585,
      "count": 6,
      "id": 9599,
      "name": "Необрастающая краска A4 T.SPEED белая 0.75 l",
      "price": 35.5445,
      "sum": 213.267,
    },
    {
      "brand": "Nautix",
      "category": "Необрастающие краски сезонные",
      "categoryId": 587,
      "count": 6,
      "id": 9751,
      "name": "Необрастающая краска Marin красная 0.75 l",
      "price": 16.0702,
      "sum": 96.4212,
    },
    {
      "brand": "Nautix",
      "category": "Эпоксидные шпаклевки",
      "categoryId": 770,
      "count": 6,
      "id": 48189,
      "name": "Шпаклевка Watertight в тубе 250 ml",
      "price": 14.0013,
      "sum": 84.0078,
    },
    {
      "brand": "Nautix",
      "category": "Растворители",
      "categoryId": 171,
      "count": 6,
      "id": 54550,
      "name": "Растворитель DE 0.75 l",
      "price": 7.0608,
      "sum": 42.3648,
    },
  ],
}
`;

exports[`Get Order Order 204453 1`] = `
{
  "order": {
    "allSum": 581.9004,
    "clientName": "_jamuuk",
    "date": "2021-08-01T10:22:00.000Z",
    "id": 204453,
    "isPaid": true,
    "paidSum": 581.9004,
    "positionCount": -14,
    "productsCount": -50,
    "productsSum": 581.9004,
    "status": "продажа",
  },
  "payments": [
    {
      "date": "2021-08-13T17:35:00.000Z",
      "id": 83842,
      "sum": 2,
      "type": "Нал",
    },
    {
      "date": "2021-08-13T17:44:00.000Z",
      "id": 83843,
      "sum": 50,
      "type": "Visa",
    },
    {
      "date": "2021-08-13T17:46:00.000Z",
      "id": 83844,
      "sum": 40,
      "type": "Нал",
    },
    {
      "date": "2021-08-13T17:53:00.000Z",
      "id": 83845,
      "sum": 76,
      "type": "Visa",
    },
    {
      "date": "2021-08-19T10:19:00.000Z",
      "id": 83869,
      "sum": 384.9004,
      "type": "Нал",
    },
    {
      "date": "2021-08-19T10:20:00.000Z",
      "id": 83870,
      "sum": 29,
      "type": "Нал",
    },
  ],
  "products": [
    {
      "brand": "Harken",
      "category": "Блоки20-30 мм",
      "categoryId": 12,
      "count": 1,
      "id": 12533,
      "name": "29mm Carbo Одинарный блок",
      "price": 16.5044,
      "sum": 16.5044,
    },
    {
      "brand": "Harken",
      "category": "Блоки палубные",
      "categoryId": 579,
      "count": 1,
      "id": 12535,
      "name": "29mm Carbo палубный блок",
      "price": 15.996,
      "sum": 15.996,
    },
    {
      "brand": "TREM",
      "category": "Помпы",
      "categoryId": 81,
      "count": 1,
      "id": 18232,
      "name": "Ручная помпа откачки воды 30л., пластик",
      "price": 26,
      "sum": 26,
    },
    {
      "brand": "Harken_Sport",
      "category": "Перчатки яхтенные",
      "categoryId": 20,
      "count": 1,
      "id": 27478,
      "name": "Перчатки Reflex – 3/4 Finger M",
      "price": 38,
      "sum": 38,
    },
    {
      "brand": "Harken_Sport",
      "category": "Перчатки яхтенные",
      "categoryId": 20,
      "count": 1,
      "id": 27481,
      "name": "Перчатки Reflex – 3/4 Finger XXL",
      "price": 38,
      "sum": 38,
    },
    {
      "brand": "TREM",
      "category": "Кранцы цилиндрические",
      "categoryId": 794,
      "count": 6,
      "id": 45257,
      "name": "Кранец белый Star G3 60x18 см",
      "price": 20,
      "sum": 120,
    },
    {
      "brand": "MarineTech",
      "category": " Скобы нержавеющие",
      "categoryId": 699,
      "count": 1,
      "id": 47888,
      "name": "Скоба укороченного типа 4mm A4",
      "price": 1.5,
      "sum": 1.5,
    },
    {
      "brand": "GottifrediMaffioli",
      "category": "Полиэстер HT для оттяжек, фалов швертботов",
      "categoryId": 51,
      "count": 7,
      "id": 47977,
      "name": "T-90 полиэстер 8мм blue",
      "price": 2.5,
      "sum": 17.5,
    },
    {
      "brand": "Optiparts",
      "category": "Жилеты страховочные 50N",
      "categoryId": 475,
      "count": 2,
      "id": 54321,
      "name": "Жилет спасательный для с боковой молнией WINDESIGN JS",
      "price": 72,
      "sum": 144,
    },
    {
      "brand": "Optiparts",
      "category": "Жилеты страховочные 50N",
      "categoryId": 475,
      "count": 1,
      "id": 54324,
      "name": "Жилет спасательный для с боковой молнией WINDESIGN S",
      "price": 70,
      "sum": 70,
    },
    {
      "brand": "Optiparts",
      "category": "Перчатки яхтенные",
      "categoryId": 20,
      "count": 1,
      "id": 54524,
      "name": "Перчатки AMARA  короткие пальцы JL",
      "price": 29,
      "sum": 29,
    },
    {
      "brand": "GottifrediMaffioli",
      "category": null,
      "categoryId": null,
      "count": 4,
      "id": 56041,
      "name": "GM 112 8MM черная",
      "price": 2.5,
      "sum": 10,
    },
    {
      "brand": "Musto",
      "category": null,
      "categoryId": null,
      "count": 1,
      "id": 56069,
      "name": "Перчатки sailing SF gloves",
      "price": 29,
      "sum": 29,
    },
    {
      "brand": "Gleinstein",
      "category": null,
      "categoryId": null,
      "count": 22,
      "id": 58085,
      "name": "Standard 6mm, white, marker red",
      "price": 1.2,
      "sum": 26.4,
    },
  ],
}
`;

exports[`Get Order Order 204426 1`] = `
{
  "order": {
    "allSum": -100,
    "clientName": "Reino",
    "date": "2020-08-06T19:38:00.000Z",
    "id": 204426,
    "isPaid": false,
    "paidSum": null,
    "positionCount": 2,
    "productsCount": 2,
    "productsSum": -100,
    "status": "покупка",
  },
  "payments": [],
  "products": [
    {
      "brand": "Луч",
      "category": "Рулевое и шверт",
      "categoryId": 111,
      "count": 1,
      "id": 30215,
      "name": "Перо руля Луч (Эстония)",
      "price": 40,
      "sum": 40,
    },
    {
      "brand": "Луч",
      "category": "Рулевое и шверт",
      "categoryId": 111,
      "count": 1,
      "id": 31575,
      "name": "Шверт Луч (Эстония)",
      "price": 60,
      "sum": 60,
    },
  ],
}
`;

exports[`Get Order Order 205162 1`] = `
{
  "order": {
    "allSum": -2799.5089,
    "clientName": "Ropebox Eesti OÜ",
    "date": "2022-08-10T15:38:00.000Z",
    "id": 205162,
    "isPaid": false,
    "paidSum": -2799.51,
    "positionCount": 30,
    "productsCount": 4074,
    "productsSum": -2799.5089,
    "status": "покупка",
  },
  "payments": [
    {
      "date": "2022-08-28T11:41:00.000Z",
      "id": 84629,
      "sum": -2799.51,
      "type": "Б/н",
    },
  ],
  "products": [
    {
      "brand": "Bainbridge-HW",
      "category": "Ткань липкая для ремонта парусов",
      "categoryId": 75,
      "count": 15,
      "id": 3099,
      "name": "Polyester Insignia RED 137 cm",
      "price": 13.2267,
      "sum": 198.4005,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ткань липкая для ремонта парусов",
      "categoryId": 75,
      "count": 5,
      "id": 3100,
      "name": "Polyester Insignia WHITE 142 cm Wide",
      "price": 14.58,
      "sum": 72.9,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Кнопки Button Snap",
      "categoryId": 497,
      "count": 5,
      "id": 3150,
      "name": "Деталь тентовой кнопки (850-2 М20)",
      "price": 13.3042,
      "sum": 66.521,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Кнопки Button Snap",
      "categoryId": 497,
      "count": 5,
      "id": 3154,
      "name": "Деталь тентовой кнопки (850-1 М20)",
      "price": 15.3553,
      "sum": 76.7765,
    },
    {
      "brand": "Bainbridge-HW",
      "category": null,
      "categoryId": null,
      "count": 100,
      "id": 3176,
      "name": "Замок молнии #10 Vislon односторонний ЧЕРНЫЙ",
      "price": 0.6528,
      "sum": 65.28,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Крючки для ползунов, карабины",
      "categoryId": 650,
      "count": 50,
      "id": 3204,
      "name": "Карабин-крючок 3,5 мм",
      "price": 0.6845,
      "sum": 34.225,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Крючки для ползунов, карабины",
      "categoryId": 650,
      "count": 100,
      "id": 3205,
      "name": "Карабин-крючок 5 мм",
      "price": 1.069,
      "sum": 106.9,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Крючки для ползунов, карабины",
      "categoryId": 650,
      "count": 50,
      "id": 3253,
      "name": "Крючок для ползуна D=10",
      "price": 0.368,
      "sum": 18.4,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Шарниры, трубы, фитинги для тентов",
      "categoryId": 485,
      "count": 100,
      "id": 3256,
      "name": "Крючок для тента, пластик, белый",
      "price": 0.144,
      "sum": 14.4,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ленты парусной ткани",
      "categoryId": 527,
      "count": 110,
      "id": 3367,
      "name": "Лента из парусной ткани 6.0Oz 100 мм",
      "price": 0.844,
      "sum": 92.84,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Люверсы",
      "categoryId": 466,
      "count": 1000,
      "id": 3407,
      "name": "Люверс никелированный разм.0, D=8.4mm",
      "price": 0.1959,
      "sum": 195.9,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Люверсы",
      "categoryId": 466,
      "count": 1000,
      "id": 3410,
      "name": "Люверс никелированный разм.2, D=11.9 мм",
      "price": 0.3554,
      "sum": 355.4,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Молнии",
      "categoryId": 157,
      "count": 9,
      "id": 3453,
      "name": "Молния 3.66 м белая с двусторонним замком",
      "price": 9.6187,
      "sum": 86.5683,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Молнии",
      "categoryId": 157,
      "count": 10,
      "id": 3455,
      "name": "Молния 3.66 м черная с двусторонним замком",
      "price": 9.3307,
      "sum": 93.307,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Молнии",
      "categoryId": 157,
      "count": 10,
      "id": 3456,
      "name": "Молния 4,57 м белая с двусторонним замком",
      "price": 11.3178,
      "sum": 113.178,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Молнии",
      "categoryId": 157,
      "count": 100,
      "id": 3462,
      "name": "Молния рулон. #10 Vislon белая",
      "price": 3.3022,
      "sum": 330.22,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Вырубка и оснастка для люверсов, кренгельсов",
      "categoryId": 528,
      "count": 1,
      "id": 3549,
      "name": "Оснастка для устан. люверсов English, разм.2",
      "price": 30.42,
      "sum": 30.42,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ползуны",
      "categoryId": 447,
      "count": 10,
      "id": 3573,
      "name": "Ползун со стальной дужкой D=8",
      "price": 3.1517,
      "sum": 31.517,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ползуны",
      "categoryId": 447,
      "count": 50,
      "id": 3575,
      "name": "Ползун пластик D=12mm",
      "price": 0.4435,
      "sum": 22.175,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ползуны",
      "categoryId": 447,
      "count": 50,
      "id": 3580,
      "name": "Ползун пластик D=9mm",
      "price": 0.3775,
      "sum": 18.875,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ползуны",
      "categoryId": 447,
      "count": 50,
      "id": 3588,
      "name": "Ползун со стальной дужкой D=6.9",
      "price": 0.9343,
      "sum": 46.715,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ползуны",
      "categoryId": 447,
      "count": 50,
      "id": 3591,
      "name": "Ползун со стальной дужкой D=11",
      "price": 1.123,
      "sum": 56.15,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Люверсы",
      "categoryId": 466,
      "count": 1000,
      "id": 9095,
      "name": "Люверс никелированный разм.00, D=6.4mm",
      "price": 0.153,
      "sum": 153,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Инструменты парусного мастера (иглы, гардаманы, ножи)",
      "categoryId": 151,
      "count": 2,
      "id": 16596,
      "name": "Гардаман N2, правая рука",
      "price": 7.4247,
      "sum": 14.8494,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Крючки для ползунов, карабины",
      "categoryId": 650,
      "count": 50,
      "id": 16607,
      "name": "Крючок для ползуна D=10",
      "price": 1.038,
      "sum": 51.9,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Ленты парусной ткани",
      "categoryId": 527,
      "count": 100,
      "id": 32414,
      "name": "Лента из парусной ткани 7.0Oz 90 мм",
      "price": 0.751,
      "sum": 75.1,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Инструменты парусного мастера (иглы, гардаманы, ножи)",
      "categoryId": 151,
      "count": 2,
      "id": 44013,
      "name": "Гардаман N3, правая рука",
      "price": 7.9522,
      "sum": 15.9044,
    },
    {
      "brand": "Bainbridge-HW",
      "category": "Инструменты парусного мастера (иглы, гардаманы, ножи)",
      "categoryId": 151,
      "count": 2,
      "id": 44014,
      "name": "Гардаман N3, левая рука",
      "price": 7.9522,
      "sum": 15.9044,
    },
    {
      "brand": "Bainbridge-SC",
      "category": null,
      "categoryId": null,
      "count": 28,
      "id": 50495,
      "name": "Ткань парусная SPХ-525F 137 cm",
      "price": 9.6278,
      "sum": 269.5784,
    },
    {
      "brand": "Bainbridge-HW",
      "category": null,
      "categoryId": null,
      "count": 10,
      "id": 53740,
      "name": "Лента для ремонта RED 50mm x 10m (замена J601RD)",
      "price": 7.6204,
      "sum": 76.204,
    },
  ],
}
`;
